stream-applications-build/pom.xml

```diff
@@ -35,12 +35,12 @@
         <nohttp-checkstyle.version>0.0.11</nohttp-checkstyle.version>
         <disable.nohttp.checks>true</disable.nohttp.checks>
         <spring-javaformat-checkstyle.version>0.0.43</spring-javaformat-checkstyle.version>
-        <spring-boot.version>3.3.6</spring-boot.version>
-        <spring.version>6.1.15</spring.version>
-        <spring-cloud.version>2023.0.4</spring-cloud.version>
-        <spring-cloud-starters.version>4.1.5</spring-cloud-starters.version>
-        <spring-cloud-function.version>4.1.4</spring-cloud-function.version>
-        <spring-cloud-stream-dependencies.version>4.1.4</spring-cloud-stream-dependencies.version>
+        <spring-boot.version>3.4.1</spring-boot.version>
+        <spring.version>6.2.1</spring.version>
+        <spring-cloud.version>2024.0.0</spring-cloud.version>
+        <spring-cloud-starters.version>4.2.0</spring-cloud-starters.version>
+        <spring-cloud-function.version>4.2.0</spring-cloud-function.version>
+        <spring-cloud-stream-dependencies.version>4.2.0</spring-cloud-stream-dependencies.version>
```

Comment:
I wonder why do we need these two if we have already Spring Cloud BOM?
Not familiar with Cloud Starter, but might be same story.